src/app/pages/listings/contents/service/listings-store.ts

```diff
@@ -0,0 +1,41 @@
+import { Injectable } from '@angular/core';
+import { ComponentStore, tapResponse } from '@ngrx/component-store';
+import { Observable } from 'rxjs';
+import { Listings } from 'src/app/models/listings';
+import { ListingsApiService } from './listings-api.service';
+
+interface ListingsState {
+  listings: Listings;
+}
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ListingsStore extends ComponentStore<ListingsState> {
+
+  readonly listings$: Observable<Listings> = this.select((state) => {
+    return state.listings;
+  });
+  readonly addListings = this.updater((state, listings: Listings) => ({
+    listings: [...state.listings, ...listings],
```

Comment:
We always want to overwrite because we get new listings every time we don't want to add existing listings.

[...state.listings, ...listings] this adds to existing listings in the store

Reply:
I think once we add loadMore button in listings page we need to add more item to existing lists.